integration-tests/test_client.py

```diff
@@ -0,0 +1,99 @@
+
+    insights_client.register()
+    assert insights_client.is_registered
+    assert oct(os.stat(file_).st_mode)[-3:] == "644"
+
+
+@pytest.mark.parametrize(
+    "file",
+    [
+        "/etc/insights-client/insights-client.motd",
+        "/etc/insights-client/insights-client.conf",
+        "/usr/lib/systemd/system/insights-client-boot.service",
+        "/usr/lib/systemd/system/insights-client-results.path",
+        "/usr/lib/systemd/system/insights-client-results.service",
+        "/usr/lib/systemd/system/insights-client.service",
+        "/usr/lib/systemd/system/insights-client.timer",
+        "/usr/share/doc/insights-client",
+        "/usr/share/doc/insights-client/file-content-redaction.yaml.example",
+        "/usr/share/doc/insights-client/file-redaction.yaml.example",
+        "/usr/share/man/man5/insights-client.conf.5.gz",
+        "/usr/share/man/man8/insights-client.8.gz",
+    ],
+)
+def test_client_rpm_mandatory_files(file):
+    """
+    Verify the existence of mandatory files for insights-client rpm
+    """
+    if (
+        pytest.rhel_major_version in ("6", "7") or pytest.centos_major_version == "7"
+    ) and "motd" in file:
+        pytest.skip("motd feature not enabled in RHEL6/7 and centos7 RPMs")
+
+    elif pytest.rhel_major_version in ("6") and "systemd" in file:
+        pytest.skip("No systemd files in RHEL6 RPMs")
+
+    else:
+        cmd = ["rpm", "-ql", "insights-client"]
+        rpm_ql_output = subprocess.run(cmd, capture_output=True, text=True)
+        assert file in rpm_ql_output.stdout
+
+
+def test_client_logfiles_mask(insights_client):
+    """Verify that files in /var/log/insights-client have the right mode: 0600
+
+    Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1955724
+    """
+    logfiles = glob.glob("/var/log/insights-client/*.log*")
+    # It is necessary to perform some command using insights-client
+    # to populate logs
+    insights_client.register()
+    for logfile in logfiles:
+        permissions = oct(os.stat(logfile).st_mode)[-3:]
+        assert (
+            permissions == "600"
+        ), f"permission on file {logfile} must be: 600, current it is: {permissions}"
+
+
+def test_client_logdir_permissions():
+    logdir_name = "/var/log/insights-client"
+    permissions = oct(os.stat(logdir_name).st_mode)[-3:]
+    assert permissions == (
+        "700"
+    ), f"permissions on dir {logdir_name} must be: 0700, current one is: {permissions}"
+
+
+def test_verify_logrotate_is_configured(insights_client):
+    """Verify that standard logrotate is configured via /etc/logrotate.d/insights-client
+    Ref : https://bugzilla.redhat.com/show_bug.cgi?id=1940267
+    """
+    logrotate_conf_file_path = "/etc/logrotate.d/insights-client"
+
+    assert os.path.exists(logrotate_conf_file_path), "logrotate is not configured"
+    f = open("/etc/logrotate.d/insights-client")
+    content = f.read()
+    logger.info(f"Content of log file is {content}")
```

Comment:
this test in practice only checks that `/etc/logrotate.d/insights-client` exists; if there is nothing more to do for it, it seems to me it'd be better to add that path to the parameters of `test_client_rpm_mandatory_files` (`/etc/logrotate.d/insights-client` is shipped as RPM file)

Reply:
hm displaying the content of the file during the test is not exactly useful, as it will be hidden inside the test logs; in the end, the content of the file is what is shipped in the package, so it should be easy to fetch its content whenever needed

my vote here would be:
- add the existence of `/etc/logrotate.d/insights-client` to `test_client_rpm_mandatory_files`
- drop this test for now
- in case there are more in-depth checks for the logrotate enablement/configuration, add a brand-new test

Reply:
I am removing this test, will cover logging the content in new test that I am working on, validating logrotate feature.